src/circuit/circuit_builder.rs

```diff
@@ -952,11 +979,28 @@ where
         // streams.
         unsafe { circuit.nodes[id.0].eval()? };
 
-        circuit.log_scheduler_event(&SchedulerEvent::eval_end(id));
+        circuit.log_scheduler_event(&SchedulerEvent::eval_end(circuit.nodes[id.0].as_ref()));
 
         Ok(())
     }
 
+    /// Evaluate closure `f` inside a new circuit region.
+    ///
+    /// A region is a logical grouping of circuit nodes.  Regions are used
```

Comment:
This is not obvious to me - you are assuming that the closure behaves in a certain way, e.g., adds new nodes that are somehow connected to each other. If it doesn't then the region has nothing to do with a group of nodes.
Another choice may be to invert this API, where node insertion could take some additional logging code to invoke before and after.